.github/workflows/gh-pages-examples.yml

```diff
@@ -0,0 +1,60 @@
+name: Build and deploy examples to GitHub Pages
+
+on:
+  push:
+    branches:
+      - master
+      - 
+jobs:
+  build-examples:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        projects: [
+            dokka-gradle-example,
+            dokka-kotlinAsJava-example,
+            dokka-library-publishing-example,
+            dokka-multiplatform-example,
+            dokka-customFormat-example
+        ]
+    steps:
+      - name: Checkout dokka
+        uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            examples_changed:
+              - 'examples/gradle/${{ matrix.projects }}/**'
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Build html
+        run: ./gradlew dokkaHtml --no-daemon --stacktrace
+        working-directory: examples/gradle/${{ matrix.projects }}
+        if: steps.filter.outputs.examples_changed == 'true'
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        if: steps.filter.outputs.examples_changed == 'true'
+        with:
+          name: ${{ matrix.projects }}
+          path: examples/gradle/${{ matrix.projects }}/build/dokka
+
+
+  deploy-examples:
```

Comment:
Will this action run every time we push to master? Or only when `build-examples` job will detect changes?

Reply:
The job runs every time we push to master. But if there are no changes the second step ```peaceiris/actions-gh-pages``` says ```nothing to push```.